lxd/device/nic_routed.go

```diff
@@ -497,3 +502,36 @@ func (d *nicRouted) ipv6HostAddress() string {
 
 	return nicRoutedIPv6GW
 }
+
+func (d *nicRouted) isUniqueWithGatewayAutoMode(instConf instance.ConfigReader) error {
+
+	if instConf.ExpandedDevices() != nil && d.hasAutoGatewayConfiguration(d.config) {
+		routedNicDeviceNames := []string{d.config["name"]}
+
+		nicDevices := instConf.ExpandedDevices().FilterByConfigProperties(map[string]string{"nictype": "routed"})
+		if len(nicDevices) >= 1 {
+			for _, curNicDevice := range nicDevices {
+				if curNicDevice["name"] != d.name &&
+					curNicDevice["nictype"] == "routed" &&
+					d.hasAutoGatewayConfiguration(curNicDevice) {
+					routedNicDeviceNames = append(routedNicDeviceNames, curNicDevice["name"])
+				}
+			}
+
+			if len(routedNicDeviceNames) > 1 {
+
+				return fmt.Errorf("Found ['%d'] existing routed nic with gateway auto configuration devices ['%s'], "+
+					"instances can't attach more than one routed nic device with gateways in auto mode. ",
+					len(routedNicDeviceNames), strings.Join(routedNicDeviceNames[:], ", "))
+			}
+		}
+	}
+
+	return nil
+}
+
+func (d *nicRouted) hasAutoGatewayConfiguration(config deviceConfig.Device) bool {
```

Comment:
@my5t3ry As you've seen in the implementation of this function, there are two gateway keys, `ipv4.gateway` and `ipv6.gateway`. And because they can be specified separately, potentially on two different NIC devices, it should be allowed that one NIC has `ipv4.gateway=auto`` and another has `ipv6.gateway=auto`, or that a single NIC has both set to `auto`. The only thing we shouldn't allow are two routed NIC devices both specifying the same key as auto.